src/support.rs

```diff
@@ -86,6 +85,26 @@ pub unsafe fn make_call_subquery_support(input: Internal) -> Internal {
     (sublink.into_pg() as *mut pg_sys::Node).internal()
 }
 
+extension_sql!(
+    "GRANT EXECUTE ON FUNCTION @extschema@.make_call_subquery_support(internal) TO prom_reader;",
+    name = "grant_execute_make_call_subquery_support_prom_reader",
+    requires = [make_call_subquery_support]
+);
+
+/// Backwards compatibility
+#[no_mangle]
+pub extern "C" fn pg_finfo_make_call_subquery_support() -> &'static pg_sys::Pg_finfo_record {
+    const V1_API: pg_sys::Pg_finfo_record = pg_sys::Pg_finfo_record { api_version: 1 };
+    &V1_API
+}
+
+#[allow(dead_code)]
```

Comment:
Shouldn't this function be marked `#[no_mangle]`? If not, and the compiler is seeing it as dead code, then it's going to be stripped from the shared library when optimized, and even if it isn't, the symbol will be impossible to reference by name (Rust symbols are mangled, so `make_call_subquery_support` will be something like `_ZN7example26make_call_subquery_support17h9fb4fbedaed4f767E` in the binary. If it's marked `#[no_mangle]`, then rustc will always include it in the final binary, and the symbol will be referenceable as `make_call_subquery_support`. You can also remove the `pub` visibility modifier in this case, since its not actually called anywhere in our rust code (AFAICT), and rustc will always include `#[no_mangle]` extern functions.

Reply:
doh. That was a total oversight. Nice catch.